### DISCUSSION TITLE
404 error when visiting single page

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#23362] [Help]
### What version of Next.js are you using?

10.0.9

### What version of Node.js are you using?

14.4.0

### What browser are you using?

Chrome

### What operating system are you using?

macOs

### How are you deploying your application?

Localhost/Vercel

### Describe the Bug

Going into the single page is giving me a 404 error. It is showing correctly on the listings page but when clicking into I get the 404 error. 

### Expected Behavior

Show details of the record

### To Reproduce

add new record to API and try to visit it.

My page is called [slug].js

```
export async function getStaticProps({ params }) {
  const response = await axios.get(
    `${process.env.NEXT_PUBLIC_API_ENDPOINT}/course?slug=${params.slug}`
  );

  return {
    props: {
      courseData: response.data[0],
    },
    revalidate: 1,
  };
}

export async function getStaticPaths() {
  const courses = await axios.get(
    `${process.env.NEXT_PUBLIC_API_ENDPOINT}/course`
  );
  return {
    fallback: false,
    paths: courses.data.map((course) => ({
      params: { slug: course.slug },
    })),
  };
}

```

**Comment by imadatyatalah** [comment]
I think the problem is from from getStaticPaths, Replace your getStaticPaths func with that one:
```jsx
export const getStaticPaths = async () => {
  return { paths: [], fallback: true };
};
```
If it worked then the problem is just from the returned data from getStaticPaths.
[learn more about fallback="true"](https://nextjs.org/docs/basic-features/data-fetching#fallback-true)
[Static generation of a large number of pages](https://static-tweet.vercel.app/)

**Reply by ghost** [reply]
Thank you, I can't tell if it works because when I change fallback to true the build fails. It is the same for when I had my version of the code as well. Building with fallback : false works but fallback: true always fails at build time. 

It just says:

Error: Export encountered errors on following paths:
        /courses/[slug]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! eaten@0.1.0 build: `next build`
npm ERR! Exit status 1

I have no idea why this is
